test/tests/functional/pbs_multi_sched.py

```diff
@@ -647,6 +647,268 @@ def test_multi_fairshare(self):
+                                                 101, 'grp1', 40)
+        self.scheds['sc1'].add_to_resource_group('pbsuser2',
+                                                 102, 'grp1', 20)
+        self.scheds['sc1'].add_to_resource_group('pbsuser3',
+                                                 201, 'grp2', 30)
+        self.scheds['sc1'].add_to_resource_group('pbsuser4',
+                                                 202, 'grp2', 10)
+        # Add entry to the resource group of multisched 'sc2'
+        #(sched_priv_sc2 at non-default location)
+        self.scheds['sc2'].add_to_resource_group('grp1', 100, 'root', 60)
+        self.scheds['sc2'].add_to_resource_group('grp2', 200, 'root', 40)
+        self.scheds['sc2'].add_to_resource_group('pbsuser1',
+                                                 101, 'grp1', 40)
+        self.scheds['sc2'].add_to_resource_group('pbsuser2',
+                                                 102, 'grp1', 20)
+        self.scheds['sc2'].add_to_resource_group('pbsuser3',
+                                                 201, 'grp2', 30)
+        self.scheds['sc2'].add_to_resource_group('pbsuser4',
+                                                 202, 'grp2', 10)
+        # Set scheduler iteration
+        sc_iter = {'scheduler_iteration': 7}
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='default')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='sc1')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='sc2')
+        # Update scheduler config file
+        formula = 'ceil(fabs(-ncpus*(foo/100.00)*sqrt(100)))'
+        sc_config = {'fair_share': 'True', 'fairshare_decay_time': '24:00:00',
+                     'fairshare_decay_factor': 0.5,
+                     'fairshare_usage_res': formula}
+        self.scheds['default'].set_sched_config(sc_config)
+        self.scheds['sc1'].set_sched_config(sc_config)
+        self.scheds['sc2'].set_sched_config(sc_config)
+        # Set scheduling false
+        sc_scheduling = {'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='default')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='sc1')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='sc2')
+        # Set resources to node
+        resc = {'resources_available.ncpus': 1,
+                'resources_available.foo': 5000}
+        self.server.manager(MGR_CMD_SET, NODE, resc, self.mom.shortname)
+        self.server.manager(MGR_CMD_SET, NODE, resc, id='vnode[0]')
+        self.server.manager(MGR_CMD_SET, NODE, resc, id='vnode[1]')
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 0},
+                            id='vnode[4]')
+
+        # submit jobs to default scheduler
+        def_attr = {'Resource_List.select': '1:ncpus=1:foo=20',
+                    'Resource_List.walltime': 4}
```

Comment:
A 4 second job could easily be missed by the scheduler.  Two cycles have to happen for this job for the usage to be accumulated.  The cycle that the job runs, and one subsequent cycle.  The scheduler_iteration is 7.  You might miss the second cycle.  I'd increase the walltime to 10s.